### PR TITLE
Too long validation message to honor package push notification settings

### PR DESCRIPTION
**Pull request by agr**
Addresses #5838.

Validation taking too long message is now only sent if package push notifications are enabled. Changed the wording in the Gallery for the push notifications to include all push-related notifications, not only "package published".

Rendering of the text:
org text
user text

We also have "short enable notice" in the options, it was left it as is:
image

### CODE REVIEW

File: src/NuGetGallery/Views/Shared/_AccountChangeNotifications.cshtml [hunk 1]
```diff
@@ -16,11 +16,12 @@
...
             + "that your organization become an owner of their package. Disabling this setting means users cannot "
             + "contact your organization for these reasons.";
-        ViewBag.NotifyPackagePushedLabel = "Notify the organization when a package is pushed to "
+        ViewBag.NotifyPackagePushedLabel = "Notify the organization when there are status updates for a package pushed to "
             + @Config.Current.Brand
             + " using its account";
-        ViewBag.NotifyPackagePushedDescription = "This setting enables notifications whenever a package is pushed using "
-            + "the organization account. We recommend to enable this setting so that you can inspect whether a package was pushed "
+        ViewBag.NotifyPackagePushedDescription = "This setting enables notifications for package push updates, i.e. whenever "
+            + "there is a pacakge pushed using the organization account or whenever there is a delay in publishing "
```
Comment (chenriksson) [inline src/NuGetGallery/Views/Shared/_AccountChangeNotifications.cshtml]: mispelling: `pacakge`

Same below

File: src/NuGetGallery.Core/Services/CoreMessageService.cs [hunk 2]
```diff
@@ -141,7 +141,7 @@ public void SendValidationTakingTooLongNotice(Package package, string packageUrl
                 mailMessage.Body = body;
                 mailMessage.From = CoreConfiguration.GalleryNoReplyAddress;
 
-                AddAllOwnersToMailMessage(package.PackageRegistration, mailMessage);
+                AddOwnersSubscribedToPackagePushedNotification(package.PackageRegistration, mailMessage);
```
Comment (chenriksson) [inline src/NuGetGallery.Core/Services/CoreMessageService.cs]: Somewhat related... looks like `MessageService.SendPackageUploadedNotice` is dead code? Maybe remove with this PR?

File: src/NuGetGallery/Views/Shared/_AccountChangeNotifications.cshtml (hunk 1 above)
Comment (chenriksson) [reply src/NuGetGallery/Views/Shared/_AccountChangeNotifications.cshtml]: We now send notification on (async) publish, not push, right? Maybe the wording should reflect that... or even better, be more generic for any package update notifications? Is this setting also used for package owner updates?

"This setting enables notifications to organizations members for package updates, i.e., package publishing status, package ownership changes, etc.

**Comment by anangaur** [comment]
@agr Do we also send a mail when a package is unlisted/relisted?

**Comment by agr** [comment]
@anangaur, I don't think we send anything on unlist/relist. I don't see any message service method related to those.

File: src/NuGetGallery/Views/Shared/_AccountChangeNotifications.cshtml [hunk 3]
```diff
@@ -16,11 +16,12 @@
             + "organization about packages that it owns using the <em>Contact Owners</em> form, or to request "
             + "that your organization become an owner of their package. Disabling this setting means users cannot "
             + "contact your organization for these reasons.";
-        ViewBag.NotifyPackagePushedLabel = "Notify the organization when a package is pushed to "
+        ViewBag.NotifyPackagePushedLabel = "Notify the organization when there are status updates for a package pushed to "
             + @Config.Current.Brand
             + " using its account";
```
[the remark this answers is not in the document]
Comment (agr) [reply src/NuGetGallery/Views/Shared/_AccountChangeNotifications.cshtml]: The "Notify the organization whenever a package's status is updated" seems to be too generic. The setting covers only status updates for package push specifically, not general status updates (like unlists/relists as @anangaur mentioned, or package locks)

**Comment by anangaur** [comment]
Proposed text:

#### Notify when package is published to NuGet Gallery
This setting enables notifications whenever a package is published to your account. We recommend to enable this setting so that you can inspect whether a package was published intentionally and to be notified if there are unexpected delays in publishing your package.

**Comment by anangaur** [comment]
Thanks @scottbommarito

File: src/NuGetGallery/Views/Shared/_AccountChangeNotifications.cshtml [hunk 4]
```diff
@@ -6,6 +6,12 @@
     ViewBag.NotifyPackagePushedEnabled = "Receiving package push notifications";
     ViewBag.NotifyPackagePushedNotEnabled = "Not receiving package push notifications";
 
+    ViewBag.NotifyPackagePushedLabel = "Notify when package is published to " + Config.Current.Brand;
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Shared/_AccountChangeNotifications.cshtml]: >Notify when package is published [](start = 40, length = 32)

`Notify when package is published...` -> either `Notify when a package is published...` or `Notify when packages are published...`

Comment (agr) [reply src/NuGetGallery/Views/Shared/_AccountChangeNotifications.cshtml]: Fixed

File: src/NuGetGallery.Core/Services/CoreMessageService.cs (hunk 2 above)
Comment (agr) [reply src/NuGetGallery.Core/Services/CoreMessageService.cs]: Removed

**Pull request merged**